src/task/worker_pool.cpp

```diff
@@ -0,0 +1,77 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// worker_pool.cpp
+//
+// Identification: src/task/worker_pool.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "include/task/worker_pool.h"
+#include <unistd.h>
+
+namespace peloton{
+namespace task{
+
+void WorkerThread::StartThread(WorkerPool* current_pool){
+  shutdown_thread_ = false;
+  worker_thread_ = std::thread(WorkerThread::PollForWork, this, current_pool);
+}
+
+void WorkerThread::PollForWork(WorkerThread* current_thread, WorkerPool* current_pool){
+  size_t empty_count = 0;
+  Task t;
+  while(!current_thread->shutdown_thread_ || !current_pool->task_queue_.IsEmpty()){
+    // poll the queue
+    if(!current_pool->task_queue_.Dequeue(t)){
+      empty_count++;
+      if (empty_count == 10){
+        empty_count = 0;
+        usleep(1);
+      }
+      continue;
+    }
+    // call the task
+    t.func_ptr_(t.func_args_);
+    empty_count = 0;
```

Comment:
I guess you wanna break dead lock or similar stuff. If so, probably using instruction is more efficient, say `mm_pause()`. We can make it more robust like this: 
```
#if defined(USE_mm_pause)
#define PAUSE _mm_pause
#else
#define PAUSE usleep
#endif
```